### PR TITLE
WIP: add org.horizon_eda.HorizonEDA

### PR DESCRIPTION
**Pull request by carrotIndustries**

### CODE REVIEW

**Comment by carrotIndustries** [comment]
bot, build org.horizon_eda.HorizonEDA

**Comment by bilelmoussaoui** [comment]
The appdata file lacks OARS tags. see https://hughsie.github.io/oars/generate.html

File: org.horizon_eda.HorizonEDA.json [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+    "app-id": "org.horizon_eda.HorizonEDA",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "3.34",
+    "sdk": "org.gnome.Sdk",
+    "command": "horizon-eda",
+    "finish-args": [
+        "--share=network",
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=fallback-x11",
```
Comment (nedrichards) [inline org.horizon_eda.HorizonEDA.json]: you shouldn't duplicate x11 and fallback-x11 sockets. One or the other.

**Comment by carrotIndustries** [comment]
bot, build org.horizon_eda.HorizonEDA

File: org.horizon_eda.HorizonEDA.json (hunk 1 above)
Comment (carrotIndustries) [reply org.horizon_eda.HorizonEDA.json]: fixed in 4d3444fb5ed94948bf6196642b02c7b02260f6c3

File: org.horizon_eda.HorizonEDA.json [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--filesystem=home",
+        "--device=dri"
+    ],
+    "modules": [
+        {
+           "name": "mm-common",
+           "cleanup": [ "/" ],
```
Comment (barthalion) [inline org.horizon_eda.HorizonEDA.json]: It should be `*`.

File: org.horizon_eda.HorizonEDA.json [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+           "sources": [
+               {
+                   "type": "archive",
+                   "url": "https://download.gnome.org/sources/mm-common/0.9/mm-common-0.9.12.tar.xz",
+                   "sha256": "ceffdcce1e5b52742884c233ec604bf6fded12eea9da077ce7a62c02c87e7c0b"
+               }
+           ]
+        },
+        {
+            "name": "sigc++",
```
Comment (barthalion) [inline org.horizon_eda.HorizonEDA.json]: I assume all these libraries leave some development headers or static libraries around. They should be cleaned up (`cleanup` can be also global).

File: org.horizon_eda.HorizonEDA.json [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "3.34",
+    "sdk": "org.gnome.Sdk",
+    "command": "horizon-eda",
+    "finish-args": [
+        "--share=network",
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--filesystem=home",
```
Comment (barthalion) [inline org.horizon_eda.HorizonEDA.json]: Does it require access to entire directories or projects are kept in single file? If the latter, application will automatically use a file portal if code uses GtkFileChooserNative.

**Comment by barthalion** [comment]
bot, build org.horizon_eda.HorizonEDA

File: org.horizon_eda.HorizonEDA.json (hunk 4 above)
Comment (carrotIndustries) [reply org.horizon_eda.HorizonEDA.json]: It requires access to entire directories. The user opens a project file, but the actual design data is stored in files that are in the same directory as the project file.

Comment (barthalion) [reply org.horizon_eda.HorizonEDA.json]: Ack, thanks!

File: org.horizon_eda.HorizonEDA.json (hunk 3 above)
Comment (barthalion) [reply org.horizon_eda.HorizonEDA.json]:
```

Comment (barthalion) [reply org.horizon_eda.HorizonEDA.json]:
```
    "cleanup": [
        "*.a",
        "*.la",
        "/include",
        "/lib/*/*.h",
        "/share/oce-0.18",
        "/share/pkgconfig"
    ],
```

**Comment by carrotIndustries** [comment]
bot, build org.horizon_eda.HorizonEDA

**Pull request closed without merging**